addons/account/models/account_move.py

```diff
@@ -1793,6 +1794,72 @@ def _build_credit_warning_message(self, record, updated_credit):
         msg += ': %s' % formatLang(self.env, updated_credit, currency_obj=record.company_id.currency_id)
         return msg
 
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    def _compute_duplicated_ref_ids(self):
+        # The computed field only appears in the move view form and thus is supposed
+        # to only be computed for one move at a time which makes the search in loop "ok"
+        move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
+        for move in self:
+            move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
+
+
+    def _fetch_duplicate_supplier_reference(self, only_posted=False):
+        moves = self.filtered(lambda m: m.is_purchase_document() and m.ref)
+        if not moves:
+            return {}
+
+        # this case deals with single move and thus avoid issue with those edited by UI
+        if len(moves) == 1:
+            duplicates = self.env["account.move"].search(
+                [
+                    ('company_id', '=', moves.company_id.id),
+                    ('move_type', '=', moves.move_type),
+                    ('commercial_partner_id', '=', moves.commercial_partner_id.id),
+                    ('ref', '=', moves.ref),
+                    ('state', '!=', 'cancel'),
+                ]
+                + ([
+                    ('invoice_date', '=', moves.invoice_date),
+                    ('state', '=', 'posted'),
+                ] if only_posted else [])
+                + ([('id', '!=', moves.ids[0])] if moves.ids else [])
+            )
+            return {moves: duplicates}
+
+        self.env["account.move"].flush([
+            "company_id",
+            "partner_id",
+            "commercial_partner_id",
+            "ref",
+            "move_type",
+            "invoice_date",
+            "state",
+        ])
```

Comment:
The api for `flush` changed recently

Reply:
I already done it as soon as I've seen the mail but I forgot to push :expressionless: